sklearn/decomposition/_nmf.py

```diff
@@ -1717,3 +1759,617 @@ def inverse_transform(self, W):
     def _n_features_out(self):
         """Number of transformed output features."""
         return self.components_.shape[0]
+
+
+class MiniBatchNMF(NMF):
+    """Mini-Batch Non-Negative Matrix Factorization (NMF).
+
+    .. versionadded:: 1.1
+
+    Find two non-negative matrices (W, H) whose product approximates the non-
```

Comment:
+1 for using "matrices with all non-negative entries".

Reply:
There's something I don't get. In your link, the definition of non-negative matrix is exactly the kind of matrices we deal with. non-negative matrix never means positive definite and I don't think there's ambiguity here

Reply:
Based on: Fevotte, C., & Idier, J. (2011). Algorithms for nonnegative matrix factorization with the beta-divergence. Neural Computation, 23(9) (see [pdf on arxiv](https://arxiv.org/pdf/1010.1763.pdf)):

> ["non-negative"] expresses nonnegativity of the entries of matrix A (and not semidefinite positiveness)

Reply:
so non-negative is the right term

Reply:
In fact, there is ambiguity, "non-negative" can also be used as an antonym for "positive-definite", as mentioned in [the Wikipedia page](https://en.wikipedia.org/wiki/Nonnegative_matrix) given above:

> While such matrices are commonly found, the term is only occasionally used due to the possible confusion with [positive-definite matrices](https://en.wikipedia.org/wiki/Positive-definite_matrix), which are different 

Hence my remark. @jeremiedbb: do you think specifying "matrices with all non-negative entries" makes sense, here?

Reply:
i agree with @jjerphan; it does not hurt to be specific in the documentation. The ambiguity exists and not all users will click the pdf link and read up more closely. 

I don't think there's such a thing as a single "right term" because there is more than one possible generalization of non-negativity to matrix spaces.

Reply:
>  "non-negative" can also be used as an antonym for "positive-definite"

That's not what is written. It's never used as such. It's just not used that much to prevent any source of confusion.

I'm ok to be explicit in the doc, and I did not argue about that. I disagree about the possible confusion. I think it's important to use the notations and definitions so that people get used to it and stop be confused by another notation (the important thing in semi-positive definite is "definite" that we never mention here)

I pushed a change where I put both the notation and its definition.